test/e2e/proxy_publicviewer_test.go

```diff
@@ -0,0 +1,210 @@
+package e2e_test
+
+import (
+	"context"
+	"testing"
+
+	"github.com/google/uuid"
+	"github.com/stretchr/testify/require"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	toolchainv1alpha1 "github.com/codeready-toolchain/api/api/v1alpha1"
+	commonauth "github.com/codeready-toolchain/toolchain-common/pkg/test/auth"
+	testconfig "github.com/codeready-toolchain/toolchain-common/pkg/test/config"
+	testspace "github.com/codeready-toolchain/toolchain-common/pkg/test/space"
+	. "github.com/codeready-toolchain/toolchain-e2e/testsupport"
+	authsupport "github.com/codeready-toolchain/toolchain-e2e/testsupport/auth"
+	testsupportspace "github.com/codeready-toolchain/toolchain-e2e/testsupport/space"
+	"github.com/codeready-toolchain/toolchain-e2e/testsupport/spacebinding"
+	"github.com/codeready-toolchain/toolchain-e2e/testsupport/wait"
+)
+
+type proxyUser struct {
+	expectedMemberCluster *wait.MemberAwaitility
+	username              string
+	token                 string
+	identityID            uuid.UUID
+	signup                *toolchainv1alpha1.UserSignup
+	compliantUsername     string
+}
+
+// tests access to community-shared spaces
+func TestProxyPublicViewer(t *testing.T) {
+	// given
+
+	// make sure everything is ready before running the actual tests
+	awaitilities := WaitForDeployments(t)
+	hostAwait := awaitilities.Host()
+	memberAwait := awaitilities.Member1()
+
+	// public viewer is enabled in ToolchainConfig
+	hostAwait.UpdateToolchainConfig(t, testconfig.PublicViewerConfig(true))
+
+	// we create a space to share and a user for 'community-user'
+	space, _, _ := testsupportspace.CreateSpace(t, awaitilities, testspace.WithTierName("appstudio"), testspace.WithSpecTargetCluster(memberAwait.ClusterName))
+	communityUser := &proxyUser{
+		expectedMemberCluster: memberAwait,
+		username:              "community-user",
+		identityID:            uuid.New(),
+	}
+	createAppStudioUser(t, awaitilities, communityUser)
```

Comment:
minor - since apparently this function is used only for creating the community user maybe we can rename it to `createAppStudioCommunityUser` ? feel free to leave it as is in case there might future use-cases that might be different.

Reply:
good idea, done in c00f15122abb7185057104da91dd426921dc3255